test_conformance/api/negative_queue.cpp

```diff
@@ -172,3 +318,223 @@ int test_negative_create_command_queue_with_properties_khr(
+    test_failure_error_ret(
+        err, CL_INVALID_COMMAND_QUEUE,
+        "clGetCommandQueueInfo should return CL_INVALID_COMMAND_QUEUE "
+        "when: "
+        "\"command_queue is not a valid command-queue\" using a "
+        "nullptr",
+        TEST_FAIL);
+
+    if (device_supports_on_device_queue(deviceID))
+    {
+        constexpr int NUM_QUEUE_PROPERTIES = 3;
+        const std::array<cl_queue_properties, NUM_QUEUE_PROPERTIES>
+            properties = { CL_QUEUE_PROPERTIES,
+                           CL_QUEUE_OUT_OF_ORDER_EXEC_MODE_ENABLE, 0 };
+        cl_int err = CL_INVALID_VALUE;
+        clCommandQueueWrapper cmd_queue = clCreateCommandQueueWithProperties(
+            context, deviceID, properties.data(), &err);
+        test_error(err, "clCreateCommandQueueWithProperties");
+        cl_uint queue_size = -1;
+        err = clGetCommandQueueInfo(cmd_queue, CL_QUEUE_SIZE,
+                                    sizeof(queue_size), &queue_size, nullptr);
+        test_failure_error_ret(err, CL_INVALID_COMMAND_QUEUE,
+                               "clGetCommandQueueInfo should return "
+                               "CL_INVALID_COMMAND_QUEUE when: \"command_queue "
+                               "is not a valid command-queue for param_name\"",
+                               TEST_FAIL);
+    }
+
+    constexpr cl_command_queue_info invalid_param = -1;
+    err = clGetCommandQueueInfo(queue, invalid_param, 0, nullptr, nullptr);
+    test_failure_error_ret(
+        err, CL_INVALID_VALUE,
+        "clGetCommandQueueInfo should return CL_INVALID_VALUE when: "
+        "\"param_name is not one of the supported values\"",
+        TEST_FAIL);
+
+
+    cl_uint ref_count = -1;
+    err = clGetCommandQueueInfo(queue, CL_QUEUE_REFERENCE_COUNT, 0, &ref_count,
+                                nullptr);
+    test_failure_error_ret(err, CL_INVALID_VALUE,
+                           "clGetCommandQueueInfo should return "
+                           "CL_INVALID_VALUE when: \"size "
+                           "in "
+                           "bytes specified by param_value_size is < "
+                           "size of return type and "
+                           "param_value is not a NULL value\"",
+                           TEST_FAIL);
+
+    return TEST_PASS;
+}
+
+int test_negative_set_command_queue_property(cl_device_id deviceID,
+                                             cl_context context,
+                                             cl_command_queue queue,
+                                             int num_elements)
+{
+    cl_queue_properties property(CL_QUEUE_OUT_OF_ORDER_EXEC_MODE_ENABLE);
+    cl_int err = clSetCommandQueueProperty(nullptr, property, CL_TRUE, nullptr);
+    if (err == CL_INVALID_OPERATION)
+    {
+        // Implementations are allowed to return an error for
+        // non-OpenCL 1.0 devices. In which case, skip the test.
+        return TEST_SKIPPED_ITSELF;
```

Comment:
OpenCL 1.0 device can also return CL_INVALID_OPERATION, how can we know if it is non-OpenCL 1.0 devices.
The case return CL_INVALID_COMMAND_QUEUE since command_queue is nullptr according to https://github.com/KhronosGroup/OpenCL-ICD-Loader/blob/main/loader/icd_dispatch_generated.c#L2521.